defold-spine/src/comp_spine_model.cpp

```diff
@@ -286,21 +295,62 @@ namespace dmSpine
 
         spAnimation* animation = spine_scene->m_Skeleton->animations[index];
 
-        component->m_AnimationId = animation_id;
-        component->m_AnimationInstance = spAnimationState_setAnimation(component->m_AnimationStateInstance, trackIndex, animation, loop);
+        if (track_index < 0)
+        {
+            dmLogError("Invalid track index %d", track_index);
+            return false;
+        }
 
-        component->m_Playback = playback;
-        component->m_AnimationInstance->timeScale = playback_rate;
-        component->m_AnimationInstance->reverse = IsReverse(playback);
-        component->m_AnimationInstance->mixDuration = blend_duration;
-        component->m_Playing = 1;
+        while (track_index >= component->m_AnimationTracks.Capacity())
+            component->m_AnimationTracks.SetCapacity(component->m_AnimationTracks.Capacity() * 2);
```

Comment:
Say someone requests track 9 and the capacity is 4. The array should get reallocated to 16 for it to fit (doubled twice in capacity). But yes, maybe we could do the while to find the correct capacity separately from the call to SetCapacity.

Reply:
I'm wondering why this is not enough?

```
if (track_index >= component->m_AnimationTracks.Capacity()) {
    component->m_AnimationTracks.SetCapacity(track_index);
}
```

Reply:
Because this will cause 4 re-allocations:
```
spine.play_anim({ track = 8 })
spine.play_anim({ track = 9 })
spine.play_anim({ track = 10 })
spine.play_anim({ track = 11 })
```

Reply:
I don't think "* 2" is a good practice. "+" is a better use imo.
You can call OffsetCapacity(capacity+4) if you'd like.